nmmo/core/env.py

```diff
@@ -131,38 +116,22 @@ def action_space(self, agent):
   ############################################################################
   # Core API
 
-  def change_task(self,
-                  new_tasks: List[Union[Tuple[Task, float], Task]],
-                  task_encoding: Optional[Dict[int, np.ndarray]] = None,
-                  embedding_size: int=16,
-                  reset: bool=True,
-                  map_id=None,
-                  seed=None,
-                  options=None):
-    """ Changes the task given to each agent
-
-    Args:
-      new_task: The task to complete and calculate rewards
-      task_encoding: A mapping from eid to encoded task
-      embedding_size: The size of each embedding
-      reset: Resets the environment
-    """
-    self._tasks = [t if isinstance(t, Tuple) else (t,1) for t in new_tasks]
-    self._task_encoding = task_encoding
-    self._task_embedding_size = embedding_size
-    if reset:
-      self.reset(map_id=map_id, seed=seed, options=options)
-
   # TODO: This doesn't conform to the PettingZoo API
   # pylint: disable=arguments-renamed
-  def reset(self, map_id=None, seed=None, options=None):
+  def reset(self, map_id=None, seed=None, options=None,
+            new_tasks: List[Task]=None,
+            make_task_fn: Callable=None,
```

Comment:
we only need make_task_fn, the caller can define a lambda without having to pass kwargs. and they can always return [] if the want to clear the tasks. there's no need to accept new_tasks

Reply:
David, you suggest that we get rid of new_tasks, a list of task instances, and go only with make_task_fn, which can be
* make_task_fn = `lambda: return make_team_tasks(teams, task_spec)`   in case of the manual curriculum example
* or make_task_fn = 'lambda: return []' 

and inside reset(), `self.tasks = make_task_fn()` is called. Am I right? 

@jsuarez5341 how would the puffer like to handle this?